integrations/anthropic/tests/test_chat_generator.py

```diff
@@ -455,6 +570,12 @@ def test_convert_streaming_chunks_to_chat_message(self):
                     "index": 1,
                     "content_block": {"type": "tool_use", "id": "toolu_123", "name": "weather", "input": {}},
                 },
+                component_info=ComponentInfo.from_component(self),
+                index=1,
+                tool_calls=[ToolCallDelta(index=1, id="toolu_123", tool_name="weather", arguments=None)],
```

Comment:
I think we talked about this, but this should be `index=0` since it's the first and only tool call in this chunk. 
```suggestion
                tool_calls=[ToolCallDelta(index=0, id="toolu_123", tool_name="weather", arguments=None)],
```

Reply:
Same goes for the rest of the chunks in this test.

Reply:
Sorry if its already discussed.
Anthropic response chunks iteratively increase the `index` for each content block regardless of text and tool calls. So here` index=1` is because its the second content block in stream that has tool call.
 To achieve what you suggest we will have to handle this on our end by separately keeping track of tool calls indexes.
 WDYT?

Reply:
> So here index=1 is because its the second content block in stream that has tool call.
[Reference](https://github.com/deepset-ai/haystack-core-integrations/blob/668dd9c5cf69aa39fbf384d89ba48ba2ff03e78e/integrations/anthropic/tests/test_chat_generator.py#L130)

Right! So the overall index of the StreamingChunk should be 1 which you have already. So `StreamingChunk.index == 1`, but you're right we should then separately keep track of the tool call indexes. Remember though the tool call index is only relevant for the number of ToolCallDeltas present in a single StreamingChunk. So if there is only ever one ToolCallDelta in a StreamingChunk then the tool call index will always be 0.

Reply:
Sounds good!